stetho/src/main/java/com/facebook/stetho/inspector/protocol/module/CSS.java

```diff
@@ -96,30 +96,7 @@ public JsonRpcResult getMatchedStylesForNode(JsonRpcPeer peer, JSONObject params
         params,
         GetMatchedStylesForNodeRequest.class);
 
-    final GetMatchedStylesForNodeResult result = new GetMatchedStylesForNodeResult();
-
-    final RuleMatch match = new RuleMatch();
-    final RuleMatch accessibilityMatch = getAccessibilityRuleMatch();
-    result.matchedCSSRules = ListUtil.newImmutableList(match, accessibilityMatch);
-
-    match.matchingSelectors = ListUtil.newImmutableList(0);
-
-    Selector selector = new Selector();
-    selector.value = "<this_element>";
-
-    CSSRule rule = new CSSRule();
-
-    rule.origin = Origin.REGULAR;
-    rule.selectorList = new SelectorList();
-
-    rule.selectorList.selectors = ListUtil.newImmutableList(selector);
-
-    rule.style = new CSSStyle();
-    rule.style.cssProperties = new ArrayList<>();
-
-    match.rule = rule;
-
-    rule.style.shorthandEntries = Collections.emptyList();
+    final Map<String, List<CSSProperty>> selectors = new HashMap<>();
```

Comment:
using a `HashMap` for temporary storage like this is actually pretty heavy weight ... is there an alternative? how many keys to we expect to have here? can we reuse/recycle, especially since the strings we put in here will probably be repeated over and over?

for instance: maybe we create and cache 1 `HashMap` in an `mSelectorsCache` field. during this method, we still use it as a way to keep track of the `ArrayList` for each selector. then, we `put(selector, null)` at the end instead of either throwing away the `HashMap` or `clear()`ing it. That way we don't constantly churn on the `HashMap.Entry` allocations

I think it's reasonable to do this in a follow-up PR as well, it doesn't have to block us right now